File: ghc-4335c07/libraries/text/benchmarks/ruby/strip_tags.rb
#!/usr/bin/env ruby

require './utils.rb'

def strip_tags(filename)
  File.open(filename, 'r:utf-8') do |file|
    str = file.read

    d = 0

    str.each_char do |c|
      d += 1 if c == '<'
      putc(if d > 0 then ' ' else c end)
      d -= 1 if c == '>'
    end
  end
end

ARGV.each do |f|
  t = benchmark { strip_tags(f) }
  STDERR.puts "#{f}: #{t}"
end

File: ghc-4335c07/libraries/text/benchmarks/python/strip_tags.py
#!/usr/bin/env python

import utils, sys

def strip_tags(filename):
    string = open(filename, encoding='utf-8').read()

    d = 0
    out = []

    for c in string:
        if c == '<': d += 1

        if d > 0:
            out += ' '
        else:
            out += c

        if c == '>': d -= 1

    print(''.join(out))

for f in sys.argv[1:]:
    t = utils.benchmark(lambda: strip_tags(f))
    sys.stderr.write('{0}: {1}\n'.format(f, t))
